### DISCUSSION TITLE
fingerprinting resistance

### DISCUSSION THREAD

**Discussion by al-caveman** [qutebrowser/qutebrowser#6073] [Q&A]
hashes of canvas and webgl fingerprints are not randomised.  any way to make them so?

go to https://coveryourtracks.eff.org to see what i mean.

**Comment by The-Compiler** [comment]
No - but you can disable them entirely using the `content.canvas_reading` and `content.webgl` settings.

**Reply by al-caveman** [reply]
added:
```python
c.content.canvas_reading = False
c.content.webgl = False
```
in config, yet, https://coveryourtracks.eff.org still shows my canvas and webgl fingerprints/hashes.  i also tried to set the same by `:set -t ....` -- all same effect:  hash still visible.

any idea what's going on?  setting not working?

`:version`
```
qutebrowser v1.14.1
Git commit: 
Backend: QtWebEngine (Chromium 83.0.4103.122)
Qt: 5.15.2

CPython: 3.8.7
PyQt: 5.15.2

sip: 4.19.24
colorama: 0.4.4
pypeg2: 2.15
jinja2: 2.11.2
pygments: 2.7.4
yaml: 5.4.1
cssutils: 1.0.2 $Id$
attr: 20.3.0
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebEngine: 5.15.2
PyQt5.QtWebKitWidgets: no
pdf.js: no
sqlite: 3.34.0
QtNetwork SSL: OpenSSL 1.1.1i  8 Dec 2020

Style: QFusionStyle
Platform plugin: xcb
OpenGL: AMD, 4.5 (Compatibility Profile) Mesa 21.0.0-rc1
Platform: Linux-5.9.13-gentoo-x86_64-x86_64-Intel-R-_Core-TM-_i5-3570K_CPU_@_3.40GHz-with-glibc2.2.5, 64bit
Linux distribution: Gentoo/Linux (gentoo)
Frozen: False
Imported from /usr/lib/python3.8/site-packages/qutebrowser
Using Python from /usr/bin/python3.8
Qt library executable path: /usr/lib64/qt5/libexec, data path: /usr/share/qt5
```

**Reply by al-caveman** [reply]
not working.  it is only causing the hash to change value.  but not disappearing entirely.

**Reply by crocket** [reply]
Brave browser randomizes canvas fingerprint and WebGL fingerprint. I think randomizing fingerprint is better than disabling canvas_reading and webgl.

According to https://browserleaks.com/canvas and https://browserleaks.com/webgl, canvas_reading and webgl can actually be disabled.

Can qutebrowser randomize canvas fingerprint and WebGL fingerprint?

**Reply by The-Compiler** [reply]
@crocket No need to post this in multiple places. Let's keep it in #7140.
